inc/commonitiltask.class.php

```diff
@@ -1748,42 +1748,50 @@ function showFormMassiveAction() {
    }
 
 
-
    /**
-    * Display tasks in homepage
-    *
-    * @since 9.2
-    *
-    * @param integer $start            Start number to display
-    * @param string  $status           The task status to filter
-    * @param boolean $showgrouptickets As we display for group defined in task or not?
+    * Get tasks list
     *
-    * @return void
+    * @return DBmysqlIterator
     */
-   static function showCentralList($start, $status = 'todo', $showgrouptickets = true) {
-      global $DB, $CFG_GLPI;
+   public static function getTaskList($status, $showgrouptickets) {
+      global $DB;
 
       $prep_req = ['SELECT' => 'id', 'FROM' => self::getTable()];
 
-      $prep_req['AND'] = [];
+      $prep_req['WHERE'] = [];
       switch ($status) {
          case "todo" : // we display the task with the status `todo`
-            $prep_req['AND']['state'] = Planning::TODO;
+            $prep_req['WHERE']['state'] = Planning::TODO;
             break;
 
       }
-      if ($showgrouptickets) {
-         if (count($_SESSION['glpigroups'])) {
-            $prep_req['AND']['groups_id_tech'] = $_SESSION['glpigroups'];
-         } else {
-            $prep_req['AND'][0] = 1;
-         }
+      if ($showgrouptickets && isset($_SESSION['glpigroups']) && count($_SESSION['glpigroups'])) {
+         $prep_req['WHERE']['groups_id_tech'] = $_SESSION['glpigroups'];
       } else {
-         $prep_req['AND']['users_id_tech'] = $_SESSION['glpiID'];
+         $prep_req['WHERE']['users_id_tech'] = $_SESSION['glpiID'];
```

Comment:
It's not good, the behavior has changed.
We need:

1/ we get the task of my groups
    a/ I have group, so add where to get my groups
    b/ I'm not in a group, we must not get tasks (that's why I use 0=1)
2/ we get task personnal, the where is ok